src/hooks/use-list-behavior.js

```diff
@@ -0,0 +1,72 @@
+import { useState, useCallback } from 'react'
+
+/**
+ * Hook that returns bind props + accessibility props for
+ * "selectable" elements
+ */
+
+const resetStyles = {
+  outline: 'none'
+}
+
+export function useListBehavior({
```

Comment:
IMO this should be a hook that is responsible for composing/combining multiple hooks that are responsible for different behaviors:
- making elements keyboard clickable
- implementing roving tabindex pattern
- implementing escape to close pattern
- implementing external click to close pattern (maybe this isn't included here)